### DISCUSSION TITLE
Release v0.12.0

### DISCUSSION THREAD

**Discussion by saubyk** [Ride-The-Lightning/RTL#917] [Announcements]
Major release rolling out Offers (BOLT12) functionality for C-Lightning users.
https://github.com/Ride-The-Lightning/RTL/releases/tag/v0.12.0

Please note that the offers functionality in this release has a critical dependency on upgrading to the latest release of [C-Lightning-REST v0.6.1](https://github.com/Ride-The-Lightning/c-lightning-REST/releases/tag/v0.6.1). Please ensure to upgrade to that version.

Release highlights below.

C-Lightning:
- Experimental feature control to enable Offers for C-Lightning users
- Create/Manage Offers
- Pay with Offers
- Bookmark paid offers for future payments
- Export offers to PDF for printing
<img width="420" alt="Screen Shot 2021-12-29 at 7 00 13 PM" src="https://user-images.githubusercontent.com/39208279/147711716-a91002e3-c8ba-46df-bafd-10448e9a91ed.png">
<img width="581" alt="Screen Shot 2021-12-29 at 6 59 21 PM" src="https://user-images.githubusercontent.com/39208279/147711717-92ef98f5-f278-4c9a-930e-7aaa30418324.png">

LND:
- Compatibility fixes for version 0.14.1

Eclair:
- Compatibility fixed for version 0.6.2

Common enhancements:
- Event handling for invoice payment
- Updated button styles on modal windows

Other technical changes/improvements:
- Websocket support
- Password added as an environment variable
- RTL server migrated to TS
- Local database added

Minor bugs and UX issues. Complete list of issues addressed available [here](https://github.com/Ride-The-Lightning/RTL/milestone/27?closed=1)

RTL version compatibility matrix:
1. C-Lightning
- version 0.10.1
- Version 0.10.2
2. ECL
- version 0.6.1
- version 0.6.2
3. LND
- version 0.13.4
- version 0.14.1
4. Loop
- version 0.14.2
- version 0.15.0

Release contributors:
@ShahanaFarooqui
@diogorsergio
@violet360
@miketwenty1
@swedishfrenchpress 
@bitbandi
@saubyk

**Comment by hazrulnizam** [comment]
Setup guide needs to be updated. Sample config file is no longer located in the `docs` folder, which does not exist under the main folder now.
